src/backend/src/integrations/slack.ts

```diff
@@ -9,49 +9,96 @@ const slack = new WebClient(process.env.SLACK_BOT_TOKEN);
  * @param message - the text content of the message being sent
  * @param link - the link for the button on the message
  * @param linkButtonText - the text for the button on the message
+ * @returns the channel id and timestamp of the created slack message
  */
 export const sendMessage = async (slackId: string, message: string, link?: string, linkButtonText?: string) => {
   const { SLACK_BOT_TOKEN } = process.env;
   if (!SLACK_BOT_TOKEN) return;
 
-  // if link and link button are provided, add the button to the message, otherwise just send the markdown block
-  const block =
-    link && linkButtonText
-      ? {
-          type: 'section',
-          text: {
-            type: 'mrkdwn',
-            text: message
-          },
-          accessory: {
-            type: 'button',
-            text: {
-              type: 'plain_text',
-              emoji: true,
-              text: linkButtonText
-            },
-            url: link
-          }
-        }
-      : {
-          type: 'section',
-          text: {
-            type: 'mrkdwn',
-            text: message
-          }
-        };
+  const block = generateSlackTextBlock(message, link, linkButtonText);
 
   try {
-    await slack.chat.postMessage({
+    const response: ChatPostMessageResponse = await slack.chat.postMessage({
       token: SLACK_BOT_TOKEN,
       channel: slackId,
       text: message,
       blocks: [block],
       unfurl_links: false
     });
+
+    return response && response.channel && response.ts ? { channelId: response.channel, ts: response.ts } : null;
```

Comment:
since you're returning null in the else case, you can just do && {your object} instead of ? {your object} : null